nomad/eval_broker.go

```diff
@@ -724,6 +754,38 @@ func (b *EvalBroker) ResumeNackTimeout(evalID, token string) error {
 	return nil
 }
 
+func (b *EvalBroker) handleAckNackLocked(eval *structs.Evaluation) {
+	if eval == nil {
+		return
+	}
+
+	tEnq, ok := b.enqueuedTime[eval.ID]
+	if !ok {
+		delete(b.dequeuedTime, eval.ID)
+		return
+	}
+
+	tDeq, ok := b.dequeuedTime[eval.ID]
+	if !ok {
+		return
+	}
```

Comment:
```suggestion
	tDeq, ok := b.dequeuedTime[eval.ID]
	if !ok {
	    delete(b.enqueuedTime, eval.ID)
		return
	}
```

Cleaning up in case there's an enqueue time but no dequeue.